execution-engine/comm/tests/test_main_purse.rs

```
@@ -0,0 +1,68 @@
+extern crate casperlabs_engine_grpc_server;
+extern crate common;
+extern crate execution_engine;
+extern crate grpc;
+extern crate shared;
+extern crate storage;
+
+use std::collections::HashMap;
+
+use common::key::Key;
+use common::value::Account;
+use test_support::WasmTestBuilder;
+
+#[allow(unused)]
+mod test_support;
+
+const GENESIS_ADDR: [u8; 32] = [6u8; 32];
+const ACCOUNT_1_ADDR: [u8; 32] = [1u8; 32];
+
+#[ignore]
+#[test]
+fn should_run_main_purse_contract_genesis_account() {
+    let mut builder = WasmTestBuilder::default();
+
+    let builder = builder.run_genesis(GENESIS_ADDR, HashMap::new());
+
+    let genesis_account: Account = {
+        let tmp = builder.clone();
+        tmp.get_genesis_account().to_owned()
+    };
+
+    builder
+        .exec_with_args(
+            GENESIS_ADDR,
+            "main_purse.wasm",
+            1,
+            genesis_account.purse_id(),
+        )
+        .expect_success()
+        .commit()
+        .expect_success();
```

I think that this call duplicates the same checks done two lines above. It validates the last execution effectsyif I recall correctly.

---

It feels unnatural to me to end a test on an action vs an assertion; my brain wants to assert something at the end of the test block. But I just checked commit and see that it asserts within itself (though it then goes on to do some state management after the assert). 

I feel like the semantics of the test_support logic would be a little better with a final assert, but obviously we probably all have opinions re: making the test_support model cleaner to discuss at some future opportunity.

removing the superfluous .expect_success() in an incoming commit.

---

I mean, you could do:
```rust
.exec(…)
.commit()
.expect_success();
```
Would that be better?

---

id rather fail fast if the exec was not successful

---

Yep, even better.